src/components/search/search.jsx

```diff
@@ -0,0 +1,249 @@
+import { useState, forwardRef, useRef, createContext, useContext } from 'react';
+import { cn } from '@/utilities/functions';
+import { Search } from 'lucide-react';
+import Loader from '../loader';
+import {
+	textSizeClassNames,
+	variantClassNames,
+	hoverClassNames,
+	focusClassNames,
+	disabledClassNames,
+	baseClassNames,
+	sizeClassNames,
+	IconClasses
+} from './styles';
+import { autoUpdate, flip, FloatingPortal, offset, size, useDismiss, useFloating, useInteractions } from '@floating-ui/react';
+
+const SearchContext = createContext();
+
+const useSearchContext = () => {
+	return useContext(SearchContext);
+};
+
+const SearchBox = forwardRef(({ className, dimension = 'sm', open = false, onOpenChange = () => { }, ...props }, ref) => {
+
+	const inputRef = useRef(null);
+
+	const { refs, floatingStyles, context } = useFloating({
+		open: open,
+		onOpenChange: onOpenChange,
+		placement: 'bottom-start',
+		whileElementsMounted: autoUpdate,
+		middleware: [
+			offset(5),
+			flip({ padding: 10 }),
+			size({
+				apply({ rects, elements, availableHeight }) {
+					Object.assign(elements.floating.style, {
+						maxHeight: availableHeight,
+						width: `${inputRef.current?.clientWidth || 0}px`, 
+						fontFamily: window.getComputedStyle(elements.reference).fontFamily, // Retain parent's font family
+					});
+				},
+				padding: 10,
+			}),
+		],
+	});
+	const dismiss = useDismiss(context);
+
+	const { getReferenceProps, getFloatingProps } = useInteractions([
+		dismiss,
+	]);
+
+	return (
+		<SearchContext.Provider value={{ dimension, open, onOpenChange, refs, floatingStyles, context, getReferenceProps, getFloatingProps }}>
+			<div
+				className={cn('searchbox-wrapper relative w-full', className)}
+				{...props}
+				ref={inputRef}
+			/>
+		</SearchContext.Provider>
+	);
+});
+
+const SearchBoxInput = forwardRef(({
+	className,
+	type = "text",
+	placeholder = 'Search...',
+	variant = 'primary',
+	disabled = false,
+	value,
+	defaultValue,
+	onChange,
+	...props
+}, ref) => {
+	const [internalValue, setInternalValue] = useState(defaultValue || '');
+	const isControlled = useRef(value !== undefined);
+	const { dimension, open, onOpenChange, refs, getReferenceProps } = useSearchContext();
+
+	const handleChange = (event) => {
+		const newValue = event.target.value;
+		if (!isControlled.current) {
+			setInternalValue(newValue);
+		}
+		onChange?.(newValue, event);
+		if (typeof onOpenChange === 'function') {
+			if (newValue.trim()) {
+				onOpenChange(true); // Open the dropdown
+			} else {
+				onOpenChange(false); // Close the dropdown
+			}
+		}
+	};
+
+	const inputValue = isControlled.current ? value : internalValue;
+
+	return (
+		<div
+			tabIndex={0}
+			ref={refs.setReference}
+			className={cn("w-full group relative flex justify-center items-center gap-2 focus-within:z-10",
```

Comment:
use gap-0.5

Reply:
Resolved this as per discussion, added custom gaps for better placements.